extractor/scripting_environment.cpp

```diff
@@ -73,13 +76,37 @@ void ScriptingEnvironment::init_lua_state(lua_State *lua_state)
     // open utility libraries string library;
     luaL_openlibs(lua_state);
 
+    const auto sourceNoOp = []
+    {
+    };
+    const auto segmentNoOp = [](const FixedPointCoordinate *, const ExternalMemoryNode *,
```

Comment:
I don't think this is needed. You can use `lua_function_exists(lua_state, "segment_function");`